js/date-picker/js/lumx.date_picker_directive.js

```diff
@@ -72,6 +83,29 @@ angular.module('lumx.date-picker', [])
             generateCalendar();
         };
 
+        // selection of year
+        $scope.selectYear = function(year) {
+
+            // update selectedDate object
+            $scope.selectedDate.date.year(year);
+            // update model
+            $scope.model = $scope.selectedDate.date.toDate();
+            // update month based on current year
+            var year_diff = year - $scope.currentSelectedYear;
+            if(year_diff < 0) {
+                $scope.month = $scope.month.subtract(12 * Math.abs(year_diff), 'month');
```

Comment:
Yep, we have to update `$scope.month`. It does not store any month number or text, but actually an object returned by **moment.js**. Hence, `generateCalendar()` actually takes the current month by passing `$scope.month` to `moment()`. If we don't update the month, it will still have month of previously selected year.